gcc/d/expr.cc

```diff
@@ -148,7 +152,9 @@ class ExprVisitor : public Visitor
 	// For operands of other types, identity is defined as being the
 	// same as equality expressions.
 	tree t1 = build_expr(e->e1);
+	t1 = maybe_make_temp(t1);
 	tree t2 = build_expr(e->e2);
+	t2 = maybe_make_temp(t2);
 	this->result_ = d_convert(build_ctype(e->type),
 				  build_boolop(code, t1, t2));
```

Comment:
Maybe the calls to `maybe_make_temp` should be moved to `build_boolop`.  I *think* that won't have any undesirable side effects. :-)